common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -1157,58 +1154,64 @@
          */
         Problem.prototype.disableAllButtonsWhileRunning = function(operationCallback, isFromCheckOperation) {
             var that = this;
-            this.enableAllButtons(false, isFromCheckOperation);
+            var allButtons = [this.resetButton, this.saveButton, this.showButton, this.hintButton, this.submitButton];
+            var initiallyEnabledButtons = allButtons.filter(function(button) {
+                return !button.attr('disabled');
+            });
+            this.enableButtons(initiallyEnabledButtons, false, isFromCheckOperation);
```

Comment:
Yes, if we wanted to keep the Coffeescript style, we would still need to follow the pattern of using apply for the functions rather than just passing the variables in directly, or else we would lose the execution context. Is there a preference to not having them declared in the `function Problem()` vs having them declared and using apply?